Perlang.Tests/ConsoleApp/ProgramTest.cs

```diff
@@ -0,0 +1,56 @@
+using System.Collections.Generic;
+using Perlang.ConsoleApp;
+using Xunit;
+
+namespace Perlang.Tests.ConsoleApp
+{
+    public class ProgramTest
+    {
+        private readonly Program subject;
+        private readonly List<string> output = new List<string>();
+
+        public ProgramTest()
+        {
+            subject = new Program(standardOutputHandler: s => output.Add(s), runtimeErrorHandler: e => throw e);
+        }
+
+        [Fact]
+        public void Run_can_run_multiple_statements_separated_by_semicolon()
+        {
+            subject.Run("var a = 42; print a;");
+
+            Assert.Equal(new List<string> {"42"}, output);
+        }
+
+        [Fact]
+        public void Run_state_persists_between_invocations()
+        {
+            subject.Run("var a = 42;");
+            subject.Run("print a;");
+
+            Assert.Equal(new List<string> {"42"}, output);
+        }
+
+        [Fact]
+        public void Run_state_does_not_persist_if_one_statement_is_invalid()
+        {
+            // TODO: This program should fail. Try to find out why.
+            subject.Run("var a = 42;");
+            subject.Run("var b = 43; x;");
+            subject.Run("print b");
+
+            Assert.Equal(new List<string> {"42"}, output);
```

Comment:
This is incredibly weird. Here is the result from running this test:

```
Perlang.Tests.ConsoleApp.ProgramTest.Run_state_does_not_persist_if_one_statement_is_invalid

Xunit.Sdk.EqualException
Assert.Equal() Failure
Expected: List<String> ["42"]
Actual:   List<String> []
   at Perlang.Tests.ConsoleApp.ProgramTest.Run_state_does_not_persist_if_one_statement_is_invalid() in /home/per/git/perlang/Perlang.Tests/ConsoleApp/ProgramTest.cs:line 42
```

No error, no printing, no nothing. :thinking:

Reply:
Fixed. The problem was that the output was going to `stdout` or `stderr. I fixed support so that all output from `Program.Run()` goes to user-provided callbacks instead, making it possible to capture it from tests.